config/routes.rb

```diff
@@ -30,6 +30,7 @@
         get "display_all_partners", :as => "display_all_partners"
         get "display_partners_new_purchases", :as => "display_partners_new_purchases"
         post '/csv', to: 'dashboard#generate_csv'
+        get '/failed_report', to: 'dashboard#generate_failed_report'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.